roundevenf isn't provided on some platforms:

https://www.gnu.org/software/gnulib/manual/html_node/roundevenf.html

We build with glibc 2.17 for linux-64, and on osx platforms we currently use
macOS SDK v11.1 or earlier. This results in a test failure in
ExecutionEngine/Interpreter/intrinsics.ll for those platforms.

This test only runs when llvm is compiled with libffi (LLVM_ENABLE_FFI=ON)

Index: llvm-project/llvm/test/ExecutionEngine/Interpreter/intrinsics.ll
===================================================================
--- llvm-project.orig/llvm/test/ExecutionEngine/Interpreter/intrinsics.ll	2023-10-31 15:14:16.496537927 -0500
+++ llvm-project/llvm/test/ExecutionEngine/Interpreter/intrinsics.ll	2023-11-30 16:04:10.282202693 -0600
@@ -13,7 +13,6 @@
 declare double @llvm.trunc.f64(double)
 declare float  @llvm.round.f32(float)
 declare double @llvm.round.f64(double)
-declare float  @llvm.roundeven.f32(float)
 declare double @llvm.roundeven.f64(double)
 declare float  @llvm.copysign.f32(float, float)
 declare double @llvm.copysign.f64(double, double)
@@ -31,7 +30,6 @@
   %trunc64 = call double @llvm.trunc.f64(double 0.000000e+00)
   %round32 = call float @llvm.round.f32(float 0.000000e+00)
   %round64 = call double @llvm.round.f64(double 0.000000e+00)
-  %roundeven32 = call float @llvm.roundeven.f32(float 0.000000e+00)
   %roundeven64 = call double @llvm.roundeven.f64(double 0.000000e+00)
   %copysign32 = call float @llvm.copysign.f32(float 0.000000e+00, float 0.000000e+00)
   %copysign64 = call double @llvm.copysign.f64(double 0.000000e+00, double 0.000000e+00)
